Dockerfile

```diff
@@ -1,56 +1,53 @@
 # syntax=docker/dockerfile:1.4
 
 ####################################################################################################
-## Prepare Workspace
+## Build Packages
 
-FROM node:18-alpine AS workspace
-WORKDIR /workspace
+FROM node:18-alpine AS builder
+WORKDIR /directus
+
+ENV NODE_OPTIONS=--max-old-space-size=8192
 
-COPY pnpm-lock.yaml .
 RUN corepack enable && corepack prepare pnpm@7.30.0 --activate
+
+COPY pnpm-lock.yaml .
 RUN pnpm fetch
 COPY . .
 RUN pnpm install --recursive --offline --frozen-lockfile
 
-####################################################################################################
-## Create Deployment
-
-FROM workspace AS pruned
-WORKDIR /workspace
-ENV NODE_OPTIONS=--max-old-space-size=8192
-
-RUN pnpm --recursive --workspace-concurrency=1 run build
-RUN pnpm --filter directus deploy --prod pruned
-RUN cd pruned \
+RUN : \
+	&& pnpm --recursive --workspace-concurrency=1 run build \
+	&& pnpm --filter directus deploy --prod dist \
+	&& cd dist \
 	&& pnpm pack \
 	&& tar -zxvf *.tgz package/package.json \
 	&& mv package/package.json package.json \
 	&& rm -r *.tgz package \
-	&& mkdir database extensions uploads
+	&& mkdir -p data/database data/extensions data/uploads \
+	;
 
 ####################################################################################################
 ## Create Production Image
 
-FROM node:18-alpine
+FROM node:18-alpine AS runtime
 
 WORKDIR /directus
 
 EXPOSE 8055
 
-ENV DB_CLIENT="sqlite3"
-ENV DB_FILENAME="/directus/database/database.sqlite"
-ENV EXTENSIONS_PATH="/directus/extensions"
-ENV STORAGE_LOCAL_ROOT="/directus/uploads"
-ENV NODE_ENV="production"
-ENV NPM_CONFIG_UPDATE_NOTIFIER="false"
-
-COPY --from=pruned --chown=node:node /workspace/pruned .
+ENV \
+	DB_CLIENT="sqlite3" \
+	DB_FILENAME="/directus/data/database/database.sqlite" \
+	EXTENSIONS_PATH="/directus/data/extensions" \
+	STORAGE_LOCAL_ROOT="/directus/data/uploads" \
+	NODE_ENV="production" \
+	NPM_CONFIG_UPDATE_NOTIFIER="false"
```

Comment:
Environment variables should be declared in a single layer.

Reply:
ha we went back and forth on this. According to Docker's docs there's no additional layers or overhead doing it split up, so I guess it's a bit of personal preference nowadays 🤷🏻

Reply:
Fair enough! Do you want it left alone?

Reply:
Nah this works 👍🏻  More consistent with the `RUN` statements that way

Reply:
The reason behind this format for `RUN` instructions
```dockerfile
RUN : \
	&& ... \
	;
```
is actually just to have cleaner Git diffs / history when changing those statements (since that way it's possible to change only the line concerned - similar to e.g. trailing commas in a JS codebase)

Consequently we could argue that specifying multiple `ENV` separately would be the more appropriate counterpart (as we can change each line independently as well)
```dockerfile
ENV ...
ENV ...
```

The `ENV` instructions the don't lead to an increase the size of the build, so this is indeed more a personal preference... Personally, I'm fine with both ways 😄